packages/pn-personafisica-webapp/src/pages/__test__/NotificationDetail.page.test.tsx

```diff
@@ -597,20 +600,130 @@ describe('NotificationDetail Page', () => {
       ...paginationData,
       page: 1,
     };
+    const paymentCache = getPaymentCache();
+    expect(paymentCache?.currentPaymentPage).toBe(paginationData.page);
 
     // intercept the next request
     const secondPagePaymentInfoRequest = paymentInfoRequest.slice(
       paginationData.page * paginationData.size,
       (paginationData.page + 1) * paginationData.size
     );
 
-    mock.onPost(NOTIFICATION_PAYMENT_INFO(), secondPagePaymentInfoRequest);
-    await waitFor(() => {
-      expect(mock.history.post).toHaveLength(2);
-    });
+    console.log(secondPagePaymentInfoRequest);
+
+    console.log(
+      paymentInfo.slice(
+        paginationData.page * paginationData.size,
+        (paginationData.page + 1) * paginationData.size
+      )
+    );
+
+    mock
+      .onPost(NOTIFICATION_PAYMENT_INFO(), secondPagePaymentInfoRequest)
+      .reply(
+        200,
+        paymentInfo.slice(
+          paginationData.page * paginationData.size,
+          (paginationData.page + 1) * paginationData.size
+        )
+      );
+    console.log(mock.history);
+    // await waitFor(() => {
+    //   expect(mock.history.post).toHaveLength(2);
+    // });
 
     // check that the other payments are shown
     const secondPageItems = result?.queryAllByTestId('pagopa-item');
     expect(secondPageItems).toHaveLength(secondPagePaymentInfoRequest.length);
   });
+
+  it('should load payments from cache when reloading the page, so it does not make the same request twice', async () => {
```

Comment:
I'm asking myself, if the caching tests should be in the reducer test file and not here. What do you think?

Reply:
Maybe yes, but in component I can simulate the refresh and check the timestamp

Reply:
That's ok